From 0f489f8eec0c02e9ed1d606a2032edccf81ef025 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Thu, 17 Oct 2024 08:59:49 +0100
Subject: [PATCH] lib/inspect-osinfo.c: Add Windows Server 2025 osinfo

The guest was found to have these inspection fields:

    type: windows
    distro: windows
    product_name: Windows Server 2025 Standard
    product_variant: Server
    version: 10.0
    arch: x86_64
    hostname: WIN-84V4KKQ30SM
    build ID: 26080
    windows_systemroot: /Windows
    windows_software_hive: /Windows/System32/config/SOFTWARE
    windows_system_hive: /Windows/System32/config/SYSTEM
    windows_current_control_set: ControlSet001

Reported-by: Ming Xie
Fixes: https://issues.redhat.com/browse/RHEL-62935
---
 lib/inspect-osinfo.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/inspect-osinfo.c b/lib/inspect-osinfo.c
index c77344f63..f792d771a 100644
--- a/lib/inspect-osinfo.c
+++ b/lib/inspect-osinfo.c
@@ -142,7 +142,9 @@ guestfs_impl_inspect_get_osinfo (guestfs_h *g, const char *root)
       switch (minor) {
       case 0:
         if (strstr (product_variant, "Server")) {
-          if (strstr (product_name, "2022"))
+          if (strstr (product_name, "2025"))
+            return safe_strdup (g, "win2k25");
+          else if (strstr (product_name, "2022"))
             return safe_strdup (g, "win2k22");
           else if (strstr (product_name, "2019"))
             return safe_strdup (g, "win2k19");
-- 
2.43.0

